apps/rest_api/test/controllers/admin/user_test.exs

```diff
@@ -163,6 +164,93 @@ defmodule RestAPI.Controllers.Admin.User do
     end
   end
 
+  describe "GET #{@show_endpoint}" do
+    setup do
+      access_token = "my-access-token"
+      claims = default_claims()
+
+      {:ok, access_token: access_token, claims: claims, user: insert!(:user)}
+    end
+
+    test "should render user identity", %{
+      conn: conn,
+      access_token: access_token,
+      claims: claims,
+      user: user
+    } do
+      username = user.username
+
+      expect(AuthenticatorMock, :validate_access_token, fn token ->
+        assert access_token == token
+        {:ok, claims}
+      end)
+
+      expect(AuthenticatorMock, :get_session, fn %{"jti" => jti} ->
+        assert claims["jti"] == jti
+        {:ok, success_session(claims)}
+      end)
+
+      expect(AuthorizerMock, :authorize_admin, fn %Plug.Conn{} -> :ok end)
+
+      expect(ResourceManagerMock, :get_identity, fn input ->
+        assert is_map(input)
+
+        {:ok,
+         %{
+           id: user.id,
+           inserted_at: NaiveDateTime.utc_now(),
+           is_admin: user.is_admin,
+           status: user.status,
+           updated_at: NaiveDateTime.utc_now(),
+           username: username
+         }}
+      end)
+
+      assert %{
+               "id" => _id,
+               "inserted_at" => _inserted_at,
+               "updated_at" => _updated_at,
+               "is_admin" => false,
+               "status" => "active",
+               "username" => ^username
+             } =
+               conn
+               |> put_req_header("authorization", "Bearer #{access_token}")
+               |> get(@show_endpoint <> "username")
+               |> json_response(201)
+    end
+
+    test "if id is not an id, should return error", %{
+      conn: conn,
+      access_token: access_token,
+      claims: claims
+    } do
+      expect(AuthenticatorMock, :validate_access_token, fn token ->
+        assert access_token == token
+        {:ok, claims}
+      end)
+
+      expect(AuthenticatorMock, :get_session, fn %{"jti" => jti} ->
+        assert claims["jti"] == jti
+        {:ok, success_session(claims)}
+      end)
+
+      expect(AuthorizerMock, :authorize_admin, fn %Plug.Conn{} -> :ok end)
+
+      expect(ResourceManagerMock, :get_identity, fn input ->
+        assert is_map(input)
+
+        {:error, :some_reason}
+      end)
+
+      assert {:error, :some_areason} =
+               conn
+               |> put_req_header("authorization", "Bearer #{access_token}")
+               |> get(@show_endpoint <> "1")
+               |> json_response(201)
+    end
+  end
+
   defp default_claims do
```

Comment:
@lcpojr hey luiz, we can define this in some authenticator help? to avoid repeat